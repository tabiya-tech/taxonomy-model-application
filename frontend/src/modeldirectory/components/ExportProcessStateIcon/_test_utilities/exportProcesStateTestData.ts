import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllExportProcessStatePermutations = (): ModelInfoTypes.ExportProcessState[] => {
  // generate all permutations of ExportProcessState status, errored, exportErrors, exportWarnings
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ExportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((exportErrors) => {
        allBooleans.forEach((exportWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              exportErrors: exportErrors,
              exportWarnings: exportWarnings,
            },
            downloadUrl: "https://example.com",
            timestamp: new Date(),
          });
        });
      });
    });
  });
  return allPermutations;
};
