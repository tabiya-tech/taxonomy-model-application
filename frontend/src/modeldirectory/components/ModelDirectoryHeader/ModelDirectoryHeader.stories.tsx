import { Meta, StoryObj } from "@storybook/react";
import { AuthContext, authContextDefaultValue } from "src/auth/AuthProvider";
import ModelDirectoryHeader, { ModelDirectoryHeaderProps } from "./ModelDirectoryHeader";

const meta: Meta<typeof ModelDirectoryHeader> = {
  title: "ModelDirectory/ModelDirectoryHeader",
  component: ModelDirectoryHeader,
  tags: ["autodocs"],
  argTypes: { onModelImport: { action: "onModelImport" } },
};

export default meta;

type Story = StoryObj<typeof ModelDirectoryHeader>;

const getUserWithModelManagerRole = (hasModelManagerRole: boolean) => ({
  ...authContextDefaultValue,
  hasRole: (role: string) => hasModelManagerRole,
});

export const Shown: Story = {
  args: {},
};

export const ShownWithModelManagerRole: Story = (args: ModelDirectoryHeaderProps) => (
  <AuthContext.Provider value={getUserWithModelManagerRole(true)}>
    <ModelDirectoryHeader {...args} />
  </AuthContext.Provider>
);
ShownWithModelManagerRole.args = {};

export const ShownWithNonModelManagerRole: Story = (args: ModelDirectoryHeaderProps) => (
  <AuthContext.Provider value={getUserWithModelManagerRole(false)}>
    <ModelDirectoryHeader {...args} />
  </AuthContext.Provider>
);
ShownWithNonModelManagerRole.args = {};
