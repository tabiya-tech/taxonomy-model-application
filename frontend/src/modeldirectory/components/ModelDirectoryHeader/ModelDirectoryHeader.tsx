import React from "react";
import { Box, Typography } from "@mui/material";
import PrimaryButton from "../../../theme/PrimaryButton/PrimaryButton";
import { AddCircleOutlined } from "@mui/icons-material";

const uniqueId = "8482f1cc-0786-423f-821e-34b6b712d78u";
export const DATA_TEST_ID = {
  MODEL_DIRECTORY_HEADER: `model-directory-header-${uniqueId}`,
  IMPORT_MODEL_BUTTON: `import-model-button-${uniqueId}`,
  MODEL_DIRECTORY_TITLE: `model-directory-title-${uniqueId}`,
};

export interface ModelDirectoryHeaderProps {
  onModelImport: () => void;
}

const ModelDirectoryHeader: React.FC<ModelDirectoryHeaderProps> = ({ onModelImport }: Readonly<ModelDirectoryHeaderProps>) => (
  <Box
    display="flex"
    width="100%"
    justifyContent="space-between"
    alignItems="center"
    data-testid={DATA_TEST_ID.MODEL_DIRECTORY_HEADER}
  >
    <Typography variant="h2" data-testid={DATA_TEST_ID.MODEL_DIRECTORY_TITLE}>
      Model Directory
    </Typography>
    <PrimaryButton
      onClick={() => onModelImport()}
      data-testid={DATA_TEST_ID.IMPORT_MODEL_BUTTON}
      startIcon={<AddCircleOutlined />}
    >
      Import Model
    </PrimaryButton>
  </Box>
);

export default ModelDirectoryHeader;
