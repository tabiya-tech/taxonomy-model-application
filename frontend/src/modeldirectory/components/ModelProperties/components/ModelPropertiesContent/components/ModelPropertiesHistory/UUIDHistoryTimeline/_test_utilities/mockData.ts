import { ModelInfoTypes } from "src//modelInfo/modelInfoTypes";

export function getOneFakeUUIDHistoryDetail(i: number): ModelInfoTypes.ModelHistory {
  return {
    UUID: "e26aca13-aa00-4346-9e83-edaf59314378" + i,
    id: "5e12f537f079fbdcb4e61bf" + i,
    localeShortCode: "ZA",
    name: "Model" + i,
    version: "V1.1." + i,
  };
}

export function getFakeUUIDHistoryDetailsArray(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => getOneFakeUUIDHistoryDetail(i));
}

export function getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => {
    const details = getOneFakeUUIDHistoryDetail(i);
    details.id = null;
    details.name = null;
    details.version = null;
    details.localeShortCode = null;
    return details;
  });
}
