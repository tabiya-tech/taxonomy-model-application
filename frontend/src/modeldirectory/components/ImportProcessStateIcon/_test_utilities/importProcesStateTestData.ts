import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllImportProcessStatePermutations = (): ModelInfoTypes.ImportProcessState[] => {
  // generate all permutations of ImportProcessState status, errored, parsingErrors, parsingWarnings
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ImportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((parsingErrors) => {
        allBooleans.forEach((parsingWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              parsingErrors: parsingErrors,
              parsingWarnings: parsingWarnings,
            },
            createdAt: new Date(allPermutations.length),
            updatedAt: new Date(allPermutations.length),
          });
        });
      });
    });
  });
  return allPermutations;
};
