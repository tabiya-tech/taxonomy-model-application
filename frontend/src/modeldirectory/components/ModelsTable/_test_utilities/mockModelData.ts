import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import ExportProcessStateAPISpecs from "api-specifications/exportProcessState";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";

import { v4 as uuidv4 } from "uuid";

import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getMockId } from "src/_test_utilities/mockMongoId";
import { getRandomLorem, getRandomString, getTestString } from "src/_test_utilities/specialCharacters";
import { faker } from "@faker-js/faker";
import { CELL_MAX_LENGTH } from "src/modeldirectory/components/ModelsTable/ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  const model = getArrayOfFakeModels(1)[0];
  model.id = getMockId(id ?? 1);
  return model;
}

export function getArrayOfFakeModels(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location
          .countryCode("alpha-3")
          .substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description:
        i % 2 === 0
          ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH)
          : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getArrayOfFakeModelsMaxLength(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export const fakeModel: ModelInfoTypes.ModelInfo = {
  UUID: "8d914eab-6f7d-4183-accc-a09f99887b39",
  UUIDHistory: ["8d914eab-6f7d-4183-accc-a09f99887b39"],
  description: "aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm",
  id: "000000000000000000000001",
  importProcessState: {
    id: "000000000000000000000001",
    result: {
      errored: true,
      parsingErrors: true,
      parsingWarnings: true,
    },
    status: ImportProcessStateEnums.Status.PENDING,
    createdAt: new Date("2023-10-18T17:35:10.571Z"),
    updatedAt: new Date("2023-10-18T17:35:12.571Z"),
  },
  exportProcessState: [
    {
      downloadUrl: "https://www.example.com/",
      id: "000000000000000000000001",
      result: {
        errored: true,
        exportErrors: true,
        exportWarnings: true,
      },
      status: ExportProcessStateAPISpecs.Enums.Status.PENDING,
      timestamp: new Date("2023-10-18T17:35:10.571Z"),
      createdAt: new Date("2023-10-18T17:35:10.571Z"),
      updatedAt: new Date("2023-10-18T17:35:12.571Z"),
    },
  ],
  locale: {
    UUID: "7c4ea00b-fbea-4a31-9c04-009c813f5f34",
    name: "Palestine",
    shortCode: "ETH",
  },
  name: "1/1 - fysmros tcgcnjbbrrev",
  path: "https://unwelcome-editorial.net/",
  releaseNotes: "Perferendis modi impedit necessitatibus a",
  released: true,
  tabiyaPath: "https://silver-vulture.biz",
  createdAt: new Date("2023-10-18T17:35:10.571Z"),
  updatedAt: new Date("2022-10-18T17:35:10.571Z"),
  version: "0.2.2",
};

export function getRandomImportStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getRandomExportStatus(id: number) {
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getOneFakeExportProcessState(i: number): ModelInfoTypes.ExportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomExportStatus(i),
    result: {
      errored: i % 2 === 0,
      exportErrors: i % 2 === 0,
      exportWarnings: i % 2 === 0,
    },
    downloadUrl: faker.internet.url(),
    timestamp: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}

export function getOneFakeImportProcessState(i: number): ModelInfoTypes.ImportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomImportStatus(i),
    result: {
      errored: i % 2 === 0,
      parsingErrors: i % 2 === 0,
      parsingWarnings: i % 2 === 0,
    },
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}
