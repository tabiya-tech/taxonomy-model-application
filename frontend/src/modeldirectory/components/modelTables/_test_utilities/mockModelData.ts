import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import {v4 as uuidv4} from "uuid";

import {ModelInfoTypes} from "src/modelInfo/modelInfoTypes";
import {getMockId} from "src/_test_utilities/mockMongoId";
import {getRandomLorem, getRandomString, getTestString} from "src/_test_utilities/specialCharacters";
import {faker} from "@faker-js/faker";
import {CELL_MAX_LENGTH} from "../ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  return getArrayOfFakeModels(id ? id : 1)[0];
}

export function getArrayOfFakeModels(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: Array.from({length: 20}, (_, i) => faker.lorem.word()).join(" ").substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location.countryCode("alpha-3").substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: i % 2 === 0 ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH) : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getArrayOfFakeModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getArrayOfFakeModelsForSorting(): ModelInfoTypes.ModelInfo[] {
  const baseModel: Omit<ModelInfoTypes.ModelInfo, 'name' | 'version' | 'createdAt' | 'updatedAt'> = {
    id: 'base',
    UUID: uuidv4(),
    previousUUID: uuidv4(),
    originUUID: uuidv4(),
    locale: {
      UUID: uuidv4(),
      name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      shortCode: faker.location.countryCode("alpha-3").substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
    },
    description: getRandomLorem(CELL_MAX_LENGTH / 2),
    released: true,
    releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
    path: faker.internet.url(),
    tabiyaPath: faker.internet.url(),
    importProcessState: {
      id: getMockId(10000),
      status: getRandomStatus(1),
      result: {
        errored: true,
        parsingErrors: true,
        parsingWarnings: true,
      }
    }
  };

  return [
    {
      ...baseModel,
      id: '1',
      name: 'ancient',
      version: '1.0.0',
      createdAt: new Date('2022-01-01T00:00:00.000Z'),
      updatedAt: new Date('2022-01-01T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '2',
      name: 'older',
      version: '1.1.0',
      createdAt: new Date('2022-01-02T00:00:00.000Z'),
      updatedAt: new Date('2022-01-02T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '3',
      name: 'old',
      version: '1.1.1',
      createdAt: new Date('2022-01-03T00:00:00.000Z'),
      updatedAt: new Date('2022-01-03T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '4',
      name: 'new',
      version: '1.1.2',
      createdAt: new Date('2022-01-04T00:00:00.000Z'),
      updatedAt: new Date('2022-01-04T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '5',
      name: 'newer',
      version: '1.1.3',
      createdAt: new Date('2022-01-05T00:00:00.000Z'),
      updatedAt: new Date('2022-01-05T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '6',
      name: 'newest',
      version: '1.2.2',
      createdAt: new Date('2022-01-06T00:00:00.000Z'),
      updatedAt: new Date('2022-01-06T00:00:00.000Z'),
    }
  ];
}


export function getRandomStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}
