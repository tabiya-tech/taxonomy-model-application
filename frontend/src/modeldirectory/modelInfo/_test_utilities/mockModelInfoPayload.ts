import { faker } from '@faker-js/faker';

import * as ModelInfo from "api-specifications/modelInfo";
import {v4 as uuidv4} from "uuid";

import {getMockId} from "src/_test_utilities/mockMongoId";
import {getRandomString, getTestString} from "src/_test_utilities/specialCharacters";

export function getMockModelInfoPayload(number: number): ModelInfo.Types.GET.Response.Payload {
  return Array.from({length: number}, (_, i) => {
    return {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomString(ModelInfo.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(ModelInfo.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(ModelInfo.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getTestString(ModelInfo.Constants.DESCRIPTION_MAX_LENGTH),
      released: false,
      releaseNotes: getTestString(ModelInfo.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfo.Constants.VERSION_MAX_LENGTH),
      // @ts-ignore
      createdAt: new Date().toISOString(),
      // @ts-ignore
      updatedAt: new Date().toISOString(),
      path: "https://foo/bar",
      tabiyaPath: "https://foo/bar/baz"
    }
  });
}

export function getFakeModelInfoPayload(number: number): ModelInfo.Types.GET.Response.Payload {
  return Array.from({length: number}, (_, i) => {
    return {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: faker.lorem.text().substring(0, ModelInfo.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country(),
        shortCode: faker.lorem.text().substring(0,  ModelInfo.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: faker.lorem.text().substring(0, ModelInfo.Constants.DESCRIPTION_MAX_LENGTH),
      released: false,
      releaseNotes: faker.lorem.text().substring(0, ModelInfo.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.lorem.text().substring(0, ModelInfo.Constants.VERSION_MAX_LENGTH),
      // @ts-ignore
      createdAt: faker.date.anytime().toISOString(),
      // @ts-ignore
      updatedAt: faker.date.anytime().toISOString(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url()
    }
  });
}