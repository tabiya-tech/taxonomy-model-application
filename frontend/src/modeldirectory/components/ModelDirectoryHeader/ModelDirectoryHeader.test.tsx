// mute the console
import "src/_test_utilities/consoleMock";

import ModelDirectoryHeader, { DATA_TEST_ID } from "./ModelDirectoryHeader";
import { render, screen } from "src/_test_utilities/test-utils";
import userEvent from "@testing-library/user-event";

describe("ModelDirectoryHeader", () => {
  beforeEach(() => {
    (console.error as jest.Mock).mockClear();
    (console.warn as jest.Mock).mockClear();
  });

  test("should render model directory header component", () => {
    // GIVEN an onModelImport callback function
    const givenOnModelImportCallback = () => {};

    // WHEN a ModelDirectoryHeader component is rendered with the given callback
    render(<ModelDirectoryHeader onModelImport={givenOnModelImportCallback} />);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND specific elements to be present in the document
    expect(screen.getByTestId(DATA_TEST_ID.MODEL_DIRECTORY_HEADER)).toBeInTheDocument();
    expect(screen.getByTestId(DATA_TEST_ID.IMPORT_MODEL_BUTTON)).toBeInTheDocument();
    expect(screen.getByTestId(DATA_TEST_ID.MODEL_DIRECTORY_TITLE)).toBeInTheDocument();
  });

  test("should call onModelImport when import button is clicked", async () => {
    // GIVEN an onModelImport callback function
    const givenOnModelImportCallback = jest.fn();

    // WHEN a ModelDirectoryHeader component is rendered with the given callback
    render(<ModelDirectoryHeader onModelImport={givenOnModelImportCallback} />);
    // AND the import button is clicked
    await userEvent.click(screen.getByTestId(DATA_TEST_ID.IMPORT_MODEL_BUTTON));

    // THEN expect the onModelImport callback to be triggered once
    expect(givenOnModelImportCallback).toHaveBeenCalledTimes(1);
  });
});
