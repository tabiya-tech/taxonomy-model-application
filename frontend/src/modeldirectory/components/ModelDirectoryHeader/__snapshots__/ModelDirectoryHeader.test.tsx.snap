// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDirectoryHeader should render model directory header component: model-directory-header-8482f1cc-0786-423f-821e-34b6b712d78u 1`] = `
<div
  class="MuiBox-root css-147x205"
  data-testid="model-directory-header-8482f1cc-0786-423f-821e-34b6b712d78u"
>
  <h2
    class="MuiTypography-root MuiTypography-h2 css-d86oa7-MuiTypography-root"
    data-testid="model-directory-title-8482f1cc-0786-423f-821e-34b6b712d78u"
  >
    Model Directory
  </h2>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation css-w9737w-MuiButtonBase-root-MuiButton-root"
    data-testid="import-model-button-8482f1cc-0786-423f-821e-34b6b712d78u"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-etbowg-MuiSvgIcon-root"
        data-testid="AddCircleOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </span>
    Import Model
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
